AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -26,6 +43,104 @@ class DoubleVideoViewController: UIViewController, CellTitled {
     }
+    
+    
+    //MARK: - UIImagePickerControllerDelegateMethod
+    func imagePickerController(_ picker: UIImagePickerController,
+                               didFinishPickingMediaWithInfo info: [String : Any]) {
+        switch info[UIImagePickerControllerMediaType] as! String {
+        case String(kUTTypeMovie):
+            if let url = info[UIImagePickerControllerReferenceURL] as? URL {
+                if self.movieURLTop == nil {
+                    self.movieURLTop = url
+                    self.movieURLTopJustUpdated = true
+                    finishAndUpdate()
+                } else if self.movieURLBottom == nil {
+                    self.movieURLBottom = url
+                    self.movieURLBottonJustUpdated = true
+                    finishAndUpdate()
+                } else if player1.rate == 0 {
+                    self.movieURLTop = url
+                    self.movieURLTopJustUpdated = true
+                    finishAndUpdate()
+                } else if player2.rate == 0 {
+                    self.movieURLBottom = url
+                    self.movieURLBottonJustUpdated = true
+                    finishAndUpdate()
+                } else {
+                    self.dismiss(animated: true, completion: nil)
+                    let alertController = UIAlertController(title: "Failed to play current video because not spot is available", message: nil , preferredStyle: UIAlertControllerStyle.alert)
+                    let okay = UIAlertAction(title: "Okay", style: .cancel, handler: nil)
+                    alertController.addAction(okay)
+                    present(alertController, animated: true, completion: nil)
+                }
+            }
+        default:
+            print("bad media")
+        }
+    }
+    
+    private func finishAndUpdate() {
+        // dismissing imagePickerController
+        dismiss(animated: true) {
+            if let url = self.movieURLTop, self.movieURLTopJustUpdated {
+                self.player1 = AVPlayer(url: url)
+                let playerController = AVPlayerViewController()
+                playerController.player = self.player1
+                
+                let playerLayer = AVPlayerLayer(player: self.player1)
+                DispatchQueue.main.async {
+                    self.videoContainerTop.layer.sublayers?.forEach { $0.removeFromSuperlayer() }
+                    self.videoContainerTop.layer.addSublayer(playerLayer)
+                    guard let sublayers = self.videoContainerTop.layer.sublayers
+                        else {
+                            return
+                    }
+                    for layer in sublayers {
+                        layer.frame = self.videoContainerTop.bounds
+                    }
+                    self.player1.play()
+                    self.view.reloadInputViews()
+                }
+                self.movieURLTopJustUpdated = false
```

Comment:
Since the URLs were optionals you could have set them to nil after use and used them as flags instead of having these ```Bool```s track along with them.